opencensus/common/schedule/__init__.py

```diff
@@ -47,6 +48,7 @@ def __init__(self, interval, function, args=None, kwargs=None, name=None):
 
     def run(self):
         wait_time = self.interval
+        atexit.register(self.function, *self.args, **self.kwargs)
```

Comment:
There are multiple classes that use the `PeriodicTask` class for workers, and not all of them want to flush on exit (for [example](https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-azure/opencensus/ext/azure/common/storage.py#L93)). Maybe have this functionality only for the classes that require this kind of behaviour (e.g. PeriodicMetricTask).

Also, should the worker itself be responsible for an `atexit` implementation? In OpenTelemetry, we have a top level class that the user interacts with [MeterProvider](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py#L508) that is responsible for `atexit` behaviour, which in turn calls the "stop" functions of workers. This makes is so that the worker itself does not have to be forced to do a certain behaviour for all usages.

Reply:
Moved the atexit handler to the run method of PeriodicMetricTask.

Reply:
Moved the handler to PeriodicMetricTask